libraries/core/src/main/java/com/paritytrading/philadelphia/FIXMessageParser.java

```diff
@@ -25,6 +25,8 @@
  */
 public class FIXMessageParser {
 
+    private final FIXConnection connection;
```

Comment:
We are keeping the connection in a field, without actually using it, only to be able to pass it through. To me that is a design smell. If this class was inlined in `FIXConnection` we would be able to pass `this`.

Reply:
I agree with the design smell.

`FIXMessageParser` has been a separate class from the beginning, not least for the reason that it is possible to write [a microbenchmark](https://github.com/paritytrading/philadelphia/blob/main/tests/perf-test/src/main/java/com/paritytrading/philadelphia/FIXMessageParserBenchmark.java) for it. In [1.1.0](https://github.com/paritytrading/philadelphia/releases/1.1.0), it became part of the public API (#51, #83). Retaining this use case would be nice.

One option would be to retain `FIXMessageListener` and `FIXMessageParser` as they are and to introduce `FIXConnectionListener` that passes a `connection` parameter to the `message` method. Then `FIXConnection` would bridge `FIXMessageListener` into `FIXConnectionListener`. It would avoid this particular design smell.

What do you think?